types/react-transition-group/Transition.d.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, HTMLProps } from "react";
+import { TransitionActions } from "react-transition-group";
+
+export type EndHandler = (node: HTMLElement, done: () => void) => void;
+export type EnterHandler = (node: HTMLElement, isAppearing: boolean) => void;
+export type ExitHandler = (node: HTMLElement) => void;
+
+export interface TransitionProps extends TransitionActions, HTMLProps<any> {
```

Comment:
This is also missing the `children` annotation. As useless as it is when the child is written in JSX, it will be useful when the child is written as a function.

`children` is also not optional here -- not passing a child will throw at runtime.

```ts
children: React.ReactElement<any> | (status: 'entering'|'entered'|'exiting'|'exited'|'unmounted', props: any) => void
```

Those `any` in the above example actually would have been a self-reference to `TransitionProps`; so this might need splitting into 2 interfaces -- a base interface with all the keys but `children`, and the actual `interface` used as props that extends the base interface and adds `children`. That's exactly how the props are passed through in the render function anyway (https://github.com/reactjs/react-transition-group/blob/v2.0.2/src/Transition.js#L269)